src/nvim/api/buffer.c

```diff
@@ -295,18 +296,10 @@ ArrayOf(String) nvim_buf_get_lines(uint64_t channel_id,
 
   size_t size = (size_t)(end - start);
 
-  init_line_array(lstate, &rv, size);
+  init_line_array(lstate, &rv, size, arena);
 
-  if (!buf_collect_lines(buf, size, (linenr_T)start, 0, (channel_id != VIML_INTERNAL_CALL), &rv,
-                         lstate, err)) {
-    goto end;
-  }
-
-end:
-  if (ERROR_SET(err)) {
-    api_free_array(rv);
-    rv.items = NULL;
-  }
```

Comment:
Being able to skip this boilerplate is a nice quality-of-life gain from the Arena pattern.